### ISSUE TITLE
lombok warning when building with 3.7.1

### ISSUE THREAD

**Issue by manofthepeace** [quarkusio/quarkus#38506]
### Describe the bug

In a project with lombok. When I build with 3.7.1 I get the following warning at build time;

`[WARNING] [io.quarkus.arc.processor.BeanArchives] Failed to index lombok.NonNull: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD`

I am nto sure what are the incidences of it.

### Expected behavior

not sure, no warning maybe. or something that would help resolving the warning

### Actual behavior

warning is shown.

### How to Reproduce?

I do not have a reproducer for that but could have one if needed.

### Output of `uname -a` or `ver`

Darwin Kernel Version 21.6.0

### Output of `java -version`

Temurin-21.0.2+13

### Quarkus version or git rev

3.7.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

maven 3.9.5

**Comment by mschorsch** [comment]
What a coincidence just today I had a similar case in Quarkus 3.6.7 in my case I got the message `Failed to index boolean: Class not found in Classloader ...`. It took me some time to find the error. In my case a Resteasy Reactive Resource was annotated with `@Produces(MediaType.APPLICATION_JSON)`, but the method had the return type `Boolean`. The problem could be solved by changing the annotation to `@Produces(MediaType.TEXT_PLAIN)`.

However, a better message would be desirable :)

**Comment by manofthepeace** [comment]
In my case the message is new and appeared when upgrading quarkus from 3.6.8 to 3.7.1. I do not think it is a coding error as lombok's `@NonNull` is simply an annotation on top of fields. I had this setup working on that project since about quarkus 2.2.

I must also say that the warning does not seem to relate to any breakage, everything seems to work just fine, native included. If the annotation would not be working the build would actually fail as lombok would not be able to generate the right ctors using those fields.

I have another much smaller project using quarkus + lombok I just upgraded it and same warning appeared as well (obviously)

**Comment by manofthepeace** [comment]
Added a [reproducer](https://github.com/manofthepeace/quarkus-lombok-warnings), that oddly does not complain about that `@NonNull` but because of the `@Generated` that it adds via the `lombok.config` file at the root of the repo.

**Comment by gsmet** [comment]
@mschorsch could you open a separate issue with your problem? With a 3.7.1 reproducer if you can reproduce it there? Thanks!

**Comment by mschorsch** [comment]
> @mschorsch could you open a separate issue with your problem? With a 3.7.1 reproducer if you can reproduce it there? Thanks!

Done https://github.com/quarkusio/quarkus/issues/38520#issue-2112059635

**Comment by gsmet** [comment]
@mkouba @Ladicek @manovotn this is a bit annoying and if it's a simple fix, it would be awesome to address it for 3.7.2.

**Comment by Ladicek** [comment]
If I recall, the Lombok annotations are class-retained, so we're probably missing some filter for runtime-only annotations somewhere. Shouldn't be hard to fix.

**Comment by Ladicek** [comment]
Interesting thing is that if I remove the dependency on `quarkus-smallrye-openapi` from the reproducer, the warning is gone.

I'll look deeper, this was just a funny observation I made when minimizing the reproducer.

**Comment by Ladicek** [comment]
Actually not a funny observation: the warning _looks like_ it comes from ArC, but it actually comes from SmallRye OpenAPI, which uses ArC's bean archive index for some kind of discovery.

**Comment by manofthepeace** [comment]
Nice observation. I must say I am surprised at all I could not easily create a reproducer with the exact error 2 of my quarkus apps are showing (the NonNull one). then I started adding to the reperoducer until it happened, and it oddly wasn't that easy. The thing that made it was addin the `lombok.config` which triggered the `@Generated` warning, but never could reproduce a minimal reproducer with the NonNull warning. 

tldr: I dumped all extension my quarkus app that has the issue had otherwise I could not reproduce

**Comment by manovotn** [comment]
I can confirm Ladislav's analysis as I just dag around this as well.

It seems to core from SR open-api [`Annotations` util class](https://github.com/smallrye/smallrye-open-api/blob/main/core/src/main/java/io/smallrye/openapi/runtime/util/Annotations.java#L122) which detects Lombok's `@Generated` and eventually will try to index it and fail.
This should be preventable by inspecting the `AnnotationInstance` and only working with annotations that are runtime visible (see `AnnotationInstance#runtimeVisible()`.

**Comment by manovotn** [comment]
Here's what I meant in a form of PR - https://github.com/smallrye/smallrye-open-api/pull/1736
With it applied, I can no longer see the WARN from the reproducer provided here.

**Comment by manofthepeace** [comment]
@manovotn I have tested your fix on both my quarkus app (`@NonNull` warning) + the reproducer (`@Generated` warning) And it's gone :)
